### PR TITLE
 Check for universal argument before paste

### PR DESCRIPTION
**Pull request by bet4it**
We should use `(interactive "*P")` in `spacemacs/evil-mc-paste-before` and `spacemacs/evil-mc-paste-after` to get the raw prefix argument.

`evil-paste-before` and `evil-paste-after` don't support plain prefix argument, so if we press <kbd>SPC u p</kbd> in `spacemacs`, we want to paste one time without indenting.
But there is already a meaning of <kbd>C-u C-y</kbd> in `emacs`, which is _put point at beginning, and mark at end_ when yanking. So it's ambiguous when we trigger <kbd>C-u C-y</kbd> in `spacemacs`.We may want to yank with the point at beginning or just yank without indenting.
We do both of them before, which looks like a misuse. After this pull request, <kbd>C-u C-y</kbd> will only yank without indenting. There may be some `emacs` users confuse with it, but if they don't work on `spacemacs-indent-sensitive-modes`, they should also be confused before.
And as there is only one `universal-argument` in `emacs`, we couldn't do something like _paste four times with auto indent disabled temporarily_. So maybe we should find another way to temporarily disable auto indent?

By the way, as `defadvice` is obsolete, I replace the use of `spacemacs|advise-commands` with `advice-add`.

### CODE REVIEW

File: layers/+spacemacs/spacemacs-evil/funcs.el [hunk 1]
```diff
@@ -61,7 +61,6 @@ Otherwise, revert to the default behavior (i.e. enable `evil-insert-state')."
       (evil-escape-mode t)
     (evil-escape-mode -1)))
 
-
```
Comment (duianto) [inline layers/+spacemacs/spacemacs-evil/funcs.el]: This is a form-feed character, it's used in several places, for example in the Spacemacs buffer, and even 2 other times in this same file `spacemacs-evil/funcs.el`.

They just look like empty lines in the github diff, but they look like this in Spacemacs:

emacs_2019-04-24_16-44-28

In other words, they should not be removed.

**Comment by duianto** [comment]
Another PR also suggests adding a universal argument to paste:
Fix pasting with universal argument #11858

How do this PR compare to that one?

**Comment by bet4it** [comment]
What has done in #11858 has already been included in this pull request.
And this pull request adds two extra changes:
1. Fix the errors related to universal argument when people enable `multiple-cursors` layer.
2. Rewrite `spacemacs|advise-commands` with `advice-add`, as `defadvice` is obsoleted.

File: layers/+spacemacs/spacemacs-evil/funcs.el (hunk 1 above)
Comment (bet4it) [reply layers/+spacemacs/spacemacs-evil/funcs.el]: Yes, I know it.
But in this file, all the other form-feed characters are placed right before a comment, and I think it means start of a new section. But this one doesn't like this, so I wonder whether we should delete it.

Comment (duianto) [reply layers/+spacemacs/spacemacs-evil/funcs.el]: You might be right, it seems logical to only add one before a comment/section. So it's probably fine to remove this one.

**Comment by duianto** [comment]
Ok then we'll prefer this one over #11858

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.

**Pull request closed without merging**